gmso/parameterization/topology_parameterizer.py

```diff
@@ -239,49 +299,74 @@ def run_parameterization(self):
-                raise ParameterizationError(
-                    f"The provided gmso topology doesn't have any subtopologies."
+            labels = self.topology.unique_site_labels(
+                self.config.match_ff_by, name_only=True
+            )
+            if not labels or labels == IndexedSet([None]):
+                # raise ParameterizationError(
+                warnings.warn(
+                    f"The provided gmso topology doesn't have any group/molecule."
                     f"Either use a single forcefield to apply to to whole topology "
-                    f"or provide an appropriate topology whose sub-topology names are "
+                    f"or provide an appropriate topology whose molecule names are "
                     f"the keys of the `forcefields` dictionary. Provided Forcefields: "
                     f"{self.forcefields}, Topology: {self.topology}"
                 )
-            for subtop in self.topology.subtops:
-                if subtop.name not in self.forcefields:
+
+            assert_no_boundary_bonds(self.topology)
+            for label in labels:
+                if label not in self.forcefields:
                     warnings.warn(
-                        f"Subtopology {subtop.name} will not be parameterized, as the forcefield to parameterize it "
+                        f"Group {group} will not be parameterized, as the forcefield to parameterize it "
                         f"is missing."
                     )  # FixMe: Will warning be enough?
                 else:
-                    assert_no_boundary_bonds(subtop)
                     typemap = self._get_atomtypes(
-                        self.get_ff(subtop.name),
-                        subtop,
+                        self.get_ff(label),
+                        self.topology,
+                        self.config.match_ff_by,
+                        label,
+                        self.config.use_molecule_info,
                         self.config.identify_connected_components,
-                        is_subtop=True,
                     )
+
                     self._parameterize(
-                        subtop,
+                        self.topology,
                         typemap,
-                        is_subtop=True,  # This will be removed from the future iterations
+                        label_type=self.config.match_ff_by,
+                        label=label,
+                        use_molecule_info=self.config.use_molecule_info,  # This will be removed from the future iterations
                     )
         else:
             typemap = self._get_atomtypes(
                 self.get_ff(),
                 self.topology,
-                self.config.identify_connected_components,
-                is_subtop=False,
+                use_molecule_info=self.config.use_molecule_info,
+                use_isomorphic_checks=self.config.identify_connected_components,
             )
             self._parameterize(
```

Comment:
This is extremely slow compared to atomtyping. My testing shows output:
```
Time to convert mbuild structure:  0.2597050666809082
Time to id connections:  1.8692350387573242
72 molecule polymer
Atomtyping polymer takes 0.32 seconds
Time to parameterize sites is 0.8496887683868408
Time to parameterize connections is 5.573479890823364
Applying parameters to polymer takes 6.42 seconds
396 molecule water
Atomtyping water takes 0.03 seconds
Time to parameterize sites is 4.5318381786346436
Time to parameterize connections is 5.929911136627197
Applying parameters to water takes 10.46 seconds
315 molecule cyclopentane
Atomtyping cyclopentane takes 0.07 seconds
Time to parameterize sites is 3.5760509967803955
Time to parameterize connections is 24.673861265182495
Applying parameters to cyclopentane takes 28.25 seconds
Time to apply forcefields:  45.55887007713318
```

This may be do to the clone methods. I will continue testing, but this is by far the biggest speed issue.